tests/test_client/tests.py

```diff
@@ -284,6 +284,18 @@ def test_follow_307_and_308_redirect(self):
                 self.assertEqual(response.request['PATH_INFO'], '/post_view/')
                 self.assertEqual(response.request['REQUEST_METHOD'], method.upper())
 
+    def test_follow_307_and_308_preserves_query_string(self):
+        methods = ('get', 'post', 'head', 'options', 'put', 'patch', 'delete', 'trace')
+        codes = (307, 308)
+        for method, code in itertools.product(methods, codes):
+            with self.subTest(method=method, code=code):
+                req_method = getattr(self.client, method)
+                response = req_method('/redirect_query_%s/' % code, data={'value': 'test'}, follow=True)
```

Comment:
I'm not sure why we pass `data` and build a query string in tests views :thinking: I would simplify this:
```python
def test_follow_307_and_308_no_get_preserves_query_string(self):
    methods = ('post', 'head', 'options', 'put', 'patch', 'delete', 'trace')
    codes = (307, 308)
    for method, code in itertools.product(methods, codes):
        with self.subTest(method=method, code=code):
            req_method = getattr(self.client, method)
            response = req_method('/redirect_query_%s/' % code, follow=True)
            self.assertRedirects(response, '/post_view/?hello=world', status_code=code)
```

and in `views.py`:

```python
def method_saving_307_query_view(request):
    return HttpResponseRedirect('/post_view/?hello=world', status=307)


def method_saving_308_query_view(request):
    return HttpResponseRedirect('/post_view/?hello=world', status=308)
```

Maybe I'm missing sth.

Reply:
Makes perfect sense, will simplify it.

Still very much a newbie at the test client so learning as I go

Reply:
@lothemar I realized that previous assertions were correct. The current tests work even without this patch. I will restore them, sorry.